cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go

```diff
@@ -103,7 +104,14 @@ func runCleanupNode(c workflow.RunData) error {
 	if certsDir != kubeadmapiv1.DefaultCertificatesDir {
 		klog.Warningf("[reset] WARNING: Cleaning a non-default certificates directory: %q\n", certsDir)
 	}
-	dirsToClean = append(dirsToClean, certsDir)
+
+	// It's possible the tmp dir is overriden by Env, but kubeadm unables to know how that Env is defined
+	// since it's defined by end-user, so just leaving the cleanup to the end-user instead in this case.
+	tempDir := path.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.TempDirForKubeadm)
+	dirsToClean = append(dirsToClean, tempDir)
+	if r.CleanupTmpDir() {
+		dirsToClean = append(dirsToClean, certsDir)
```

Comment:
shouldn't we always append certs dir and optionally clean the tmp dir?

Reply:
aha, my fault :)

Reply:
Done